### PR TITLE
Rewrite: CreateChannel modal component

### PR DESCRIPTION
**Pull request by tiagoevanp**
## Proposed changes (including videos or screenshots)

image

### CODE REVIEW

File: client/sidebar/header/CreateChannel.js [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+import { useForm } from '../../hooks/useForm';
+import { useEndpointActionExperimental } from '../../hooks/useEndpointAction';
+import UserAutoCompleteMultiple from '../../../ee/client/audit/UserAutoCompleteMultiple';
+import { useSetting } from '../../contexts/SettingsContext';
+import { usePermission } from '../../contexts/AuthorizationContext';
+import { useMethod } from '../../contexts/ServerContext';
+import { useComponentDidUpdate } from '../../hooks/useComponentDidUpdate';
+
+
+export const CreateChannel = ({
```
Comment (gabriellsh) [inline client/sidebar/header/CreateChannel.js:16]: Thanks for making the component generic!!! It'll help a lot on a future task.

**Review by gabriellsh — CHANGES_REQUESTED** [review]
Only thing I noticed is that whenever I create a new channel, if I mark it as public && broadcast, it creates a readonly channel, but not a broadcast one. If I create it as a private group, it is created as broadcast.

File: client/sidebar/header/CreateChannel.js (hunk 1 above)
Comment (tiagoevanp) [reply client/sidebar/header/CreateChannel.js:16]: thanks man, congratulations from the team: me and @MartinSchoeler

**Review by gabriellsh — DISMISSED** [review]

**Review by ggazzo — APPROVED** [review]

**Comment by yash-rajpal** [comment]
@ggazzo @tiagoevanp 
This PR deleted a meteor template named `tag` , this template is being used when creating direct messages and discussions.

Please see issue #20771 

Please review PR #20772 , this PR adds back this template.
Thanks :)

**Pull request merged**